File: backend-node/internal/app/response.js
/**
 * @param {import('express').Response} res
 * @param {number} ts
 * @param {*} data
 */
const success = (res, ts, data) => {
  res.status(200).json({
    ok: true,
    msg: "success",
    data: data,
    ts: ts,
  });
};

/**
 * @param {import('express').Response} res
 * @param {number} ts
 * @param {number} code
 * @param {Error} err F
 */
const error = (res, ts, code, err) => {
  res.status(code).json({
    ok: false,
    msg: "error",
    data: err.message,
    ts: ts,
  });
};

module.exports = {
  success,
  error,
};

File: backend-go/internal/app/response.go
package app

import (
	"net/http"

	"github.com/gin-gonic/gin"
)

type Response struct {
	Ok   bool        `json:"ok"`
	Msg  string      `json:"msg"`
	Data interface{} `json:"data"`
	Ts   int64       `json:"ts"`
}

func Success(c *gin.Context, ts int64, data interface{}) {
	c.JSON(http.StatusOK, Response{
		Ok:   true,
		Msg:  "success",
		Data: data,
		Ts:   ts,
	})
}

func Error(c *gin.Context, ts int64, code int, err error) {
	c.JSON(code, Response{
		Ok:   false,
		Msg:  err.Error(),
		Data: err.Error(),
		Ts:   ts,
	})
}
